File: ProblemSet_18xx/Problem_1877/solution.py
from typing import List


class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        nums.sort()
        result = 0
        
        for i in range(len(nums) // 2):
            result = max(result, nums[i] + nums[len(nums)-1-i])
        
        return result

s = Solution()
print("Result", s.minPairSum([3,5,4,2,4,6]))

File: ProblemSet_18xx/Problem_1877/solution.cs
public class Solution {
    public int MinPairSum(int[] nums) {
        Array.Sort(nums);
        int max = int.MinValue;
        for (int i = 0; i < nums.Length / 2; i++)
        {
            max = Math.Max(max, nums[i] + nums[^(i + 1)]);
        }
        return min;
    }
}